### ISSUE TITLE
disabilities_file doesn't get created if parent directory is missing

### ISSUE THREAD

**Issue by deeTEEcee** [julien6387/supvisors#112]
From what I can tell, the disabilities_file only gets generated once we run `enable` or `disable` but I noticed the full directory needs to also exist. Otherwise, it never gets created but it also doesn't throw an error.

**Comment by julien6387** [comment]
True. At the moment, the file is serialized only subsequently to a call to `enable` or `disable`.
However, there is an initial check at startup and there should be an error raised if the folder does not exist.
You should get something like this in the log traces:

> 2023-05-18 18:53:20,764;ERRO;SupvisorsOptions.get_value: The directory named as part of the path /dummy_path/disabilities.json does not exist

When it happens, the path is not stored, so calls to `enable` or `disable` would work but nothing would be persisted afterwards.

Supvisors will be updated so that:
1) the file gets written when entering the OPERATIONAL state ;
2) the file path is re-evaluated every time the file needs to be written, providing eventually the above error, and allowing the user to create it on-the-fly if missing.
The folder will not be created if it does not exist to keep consistency with Supervisor behavior.

**Comment by deeTEEcee** [comment]
thanks for the clear description! that makes sense to me. i was just curious, is there any reason why we can't just auto-create the full path in supvisor if it doesn't exist?

**Comment by julien6387** [comment]
While implementing it, I've changed my mind a bit.
The default file is written down at startup. No re-evaluation of the path after the options have been loaded but the folder tree will be created if missing (unless write permission is denied, which will provide an ERROR log trace).
